### DISCUSSION TITLE
Compiling

### DISCUSSION THREAD

**Discussion by eltikia** [eltikia/Perl-1.0#8] [Q&A] [answered]
How to compile it?

**Comment by Stagyrite** [comment, marked as answer]
Start by installing the required packages ('gcc', 'yacc' and 'make') in a target system. It might not compile when you have an old version of make. If you're on Cygwin, check that you have GNU Make 4.4.1 or so.

Fetch the files from [GitHub](https://github.com/kironia/Perl-1.0). The project can't built when sources have CRLF line endings. Either don't fetch with CRLF line endings or convert line endings to LF. 

```
git config --global core.autocrlf false
git init
git remote add origin https://github.com/Stagyrite/Perl-1.0.git
git pull origin main
```

Now you can compile a 'perl' program.

```
./Configure
make depend
make
```

The same can be done in the 'x2p' subdirectory. 

```
cd x2p
cp ../Configure .
./Configure
make depend
make
```

One might want to run all tests from the 't' directory, as those all pass, at least on Cygwin.
